lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
I'm not sure if we currently have a way to set default flags in the compiler class. We've always just asked users to set those things in their personal `compilers.yaml`. Is this something you think everyone using the `xl_r` compiler would want to have? Or is it something you just happen to use yourself?